archinstall/lib/disk/filesystem.py

```diff
@@ -167,17 +167,21 @@ def add_partition(self, partition_type, start, end, partition_format=None):
 			parted_string = f'{self.blockdevice.device} mkpart {partition_type} {start} {end}'
 
 		if self.parted(parted_string):
-			start_wait = time.time()
-
-			while previous_partition_uuids == {partition.uuid for partition in self.blockdevice.partitions.values()}:
-				if time.time() - start_wait > 10:
-					raise DiskError(f"New partition never showed up after adding new partition on {self} (timeout 10 seconds).")
-				time.sleep(0.025)
-
-			# Todo: Find a better way to detect if the new UUID of the partition has showed up.
-			#       But this will address (among other issues)
-			time.sleep(float(storage['arguments'].get('disk-sleep', 2.0))) # Let the kernel catch up with quick block devices (nvme for instance)
-			return self.blockdevice.get_partition(uuid=(previous_partition_uuids ^ {partition.uuid for partition in self.blockdevice.partitions.values()}).pop())
+			count = 0
+			while count < 10:
+				new_uuid = None
+				new_uuid_set = (previous_partition_uuids ^ {partition.uuid for partition in self.blockdevice.partitions.values()})
+				if len(new_uuid_set) > 0:
+					new_uuid = new_uuid_set.pop()
```

Comment:
Are we sure the new `PARTUUID` is always at the end? Are they sorted by time/order or by name?

Reply:
TBH is the part i got from the original code, so i have no idea. 
The trick is that at the start of the method the existing partition list is fetched (Blockdevices.partitions is a dynamic property), so only the  UUIDs from partitions created while the method executes should appear at new_uuid_set.
Noone expects someone doing partitioning in parallel, so the code works ... almost always.
OTOH it is something worth investigating to get the code "concurrency safe".

Reply:
Sadly not possible without a serious rewrite (and probably even then). The only chance I know to avoid collisions is to compare the partition.path, which is not available at this point of time for the just created partition.  To get the path  one should check parted print or lsblk for a partition with the same characteristics as the one created and not existing previously. As unsafe for concurrency as our actual code, as collision prone, and probably a coding nightmare. 
Better a clearly documented warning as not to try to partition in parallel (which is simply common sense) than adding that much complexity

Reply:
Worst part is that I wrote that code, and apparently I forgot I did and I forgot the trick I used. But you've correctly identified the intended behavior of that pop and I thank you for explaining it to me, even tho I created it haha.